src/server/pfs/server/obj_block_api_server.go

```diff
@@ -534,7 +544,10 @@ func (s *objBlockAPIServer) InspectObject(ctx context.Context, request *pfsclien
 }
 
 func (s *objBlockAPIServer) CheckObject(ctx context.Context, request *pfsclient.CheckObjectRequest) (response *pfsclient.CheckObjectResponse, retErr error) {
-	func() { s.Log(request, nil, nil, 0) }()
+	func() {
+		tracing.TagAnySpan(ctx, "err", retErr, "object", s.prettyObjPath(request.Object))
+		s.Log(request, nil, nil, 0)
```

Comment:
Do we want to log out time here? (or take out time logging above if it should appear in the trace native?)

Reply:
This was actually a mistake on my part, in where I put the tracing—though good catch noticing the discrepancy!

The way that `s.Log()` works is that when it's called it inspects its own stack, goes up the stack a certain number of frames, extracts the caller's name from that ancestral stack frame, and then uses the caller's name in its logging. Because the number of stack frames that it traverses is fixed, and we always call `defer func() { ... s.Log() }`, we need to wrap the logging that we do at the *beginning* of a call in a `func() { ... }` as well so that there are the same number of stack frames between `CheckObject` and `s.Log` in all cases.

So, basically, this function runs at the beginning of the call (note no `defer`). I meant to put `tracing.TagAnySpan` in the func right below that, that *is* deferred, so I've now moved it.

Reply:
(I attempted to find any other places where I might have made the same mistake, and couldn't, but if you see any let me know!)